espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/model/SourceDocument.java

```diff
@@ -0,0 +1,138 @@
+import javax.xml.transform.stream.StreamResult;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+
+import io.appium.espressoserver.lib.handlers.exceptions.XPathLookupException;
+import io.appium.espressoserver.lib.viewaction.RootViewFinder;
+
+public class SourceDocument {
+
+    private Document doc;
+    private Map<Element, View> viewMap = new HashMap<>();
+    private boolean cacheElementReferences;
+    private final DocumentBuilderFactory docFactory = DocumentBuilderFactory.newInstance();
+    private final DocumentBuilder docBuilder = docFactory.newDocumentBuilder();
+    private TransformerFactory transformerFactory = TransformerFactory.newInstance();
+    private Transformer transformer;
+
+    public SourceDocument() throws ParserConfigurationException, TransformerException {
+        init();
+        buildXML(false);
+    }
+
+    private SourceDocument(boolean cacheElementReferences) throws ParserConfigurationException, TransformerException {
+        init();
+        buildXML(cacheElementReferences);
+    }
+
+    private void init() throws TransformerException {
+        transformer = transformerFactory.newTransformer();
+        transformer.setOutputProperty(OutputKeys.INDENT, "yes");
+        transformer.setOutputProperty("{http://xml.apache.org/xslt}indent-amount", "2");
+    }
+    /**
+     * Recursively visit all of the views and map them to XML elements
+     * @param cacheElementReferences Keep references of element's mapped to views
+     * @throws ParserConfigurationException
+     */
+    private void buildXML(boolean cacheElementReferences) throws ParserConfigurationException {
+        // Create an empty document
+        this.cacheElementReferences = cacheElementReferences;
+        doc = docBuilder.newDocument();
+
+        // Get reference to root view
+        View rootView = (new RootViewFinder()).getRootView();
+        buildXML(doc, null, rootView);
+    }
+
+    /**
+     * Recursively visit all of the views and map them to XML elements
+     * @param doc XML Document
+     * @param parentElement Element that this new element will be appended to
+     * @param view Android View that will map to an Element
+     */
+    private void buildXML(Document doc, Element parentElement, View view) {
+        // TODO: Investigate if this is safe and if we need to strip out characters before constructing XML document
+        Element element = doc.createElement(view.getClass().getName());
+
+        // Set attributes
+        ViewElement viewElement = new ViewElement(view);
+        element.setAttribute(ViewAttributesEnum.CONTENT_DESC.getName(), viewElement.getContentDescription().toString());
+        element.setAttribute(ViewAttributesEnum.BOUNDS.getName(), viewElement.getBounds().toShortString());
+        element.setAttribute(ViewAttributesEnum.FOCUSED.getName(), viewElement.isFocused() ? "true" : "false");
```

Comment:
any chance to return **Boolean** type and call toString() instead of ` ? "true" : "false"`?